packages/table/src/TableBody.tsx

```diff
@@ -1,17 +1,57 @@
-import React from 'react';
-import { useTableContext } from './TableContext';
+import React, { useMemo, useCallback } from 'react';
+import { getDataComparisonFunction, useTableContext } from './TableContext';
 import { TableProps } from './Table';
 
 type TableBodyProps<Shape> = Pick<TableProps<Shape>, 'children'>;
 
+/**
+ * Hook that avoids re-rendering children when the data used to
+ * render them hasn't changed, even if their ordering has changed.
+ */
+function useRenderedChildren<Props>(
+  propList: Array<Props>,
+  renderFunction: (props: Props) => React.ReactNode,
+  compareFn?: (a: Props, b: Props) => number,
+): Array<React.ReactNode> {
+  const pairs: Array<[Props, React.ReactNode]> = useMemo(
+    () =>
+      propList.map((props, index) => [
+        props,
+        <React.Fragment key={index}>{renderFunction(props)}</React.Fragment>,
+      ]),
+    [propList, renderFunction],
+  );
+
+  return useMemo(() => {
+    if (!compareFn) {
+      return pairs.map(([_, renderResult]) => renderResult);
+    }
+
+    return pairs
+      .sort(([a], [b]) => compareFn(a, b))
+      .map(([_, renderResult]) => renderResult);
+  }, [pairs, compareFn]);
+}
+
 function TableBody<Shape>({ children }: TableBodyProps<Shape>) {
   const {
-    state: { data },
+    state: { data, sort },
   } = useTableContext();
+  let compareFn: ((a: Shape, b: Shape) => number) | undefined;
 
-  return (
-    <tbody>{data.map((datum, index) => children({ datum, index }))}</tbody>
-  );
+  if (sort) {
+    const { direction, accessorValue } = sort;
+
+    if (direction && accessorValue) {
+      compareFn = getDataComparisonFunction({ direction, accessorValue });
+    }
+  }
+
+  const renderFunction = useCallback(datum => children({ datum }), [children]);
```

Comment:
Does this do anything? Isn't children always going to be a new instance as we instruct users to pass in an inline function?

Reply:
children doesn't change if the parent doesn't re-render. changing the sort order doesn't cause the parent to re-render.

Reply:
In practice this is changing on every render I imagine? This is to catch those instances where we can?

Does this make a material difference if it's removed?

Reply:
no, in general it will not change on every render as I already explained and yes, it makes the difference between re-rendering every row and not.

Reply:
For a contrived example:

```jsx
function MyPage() {
  const childrenFn = () => {};
  return (
    <Button onClick={increment} />
    <Table>{childrenFn}</Table>
  )
}
```

If you click `increment` that'll cause `MyPage` to render and `childrenFn` to be re-created, thus resulting in a new instance and needing to call this `useCallback` hook written here, right? Just making sure we're aligned on understanding.

Reply:
in this example `MyPage` is rerendered and `childrenFn` is recreated because the counter state belongs to `MyPage`. the data in a table belongs to the table's context and changing the sort order is also belongs to the table's context (and doesn't even touch the row data).

Reply:
Aha! So this is specific for the sorting case, which yes, is a great one to cover. Ty for venturing down this explanation path!